src/ru/nsu/ccfit/zuev/osu/MainActivity.java

```diff
@@ -298,13 +297,39 @@ public void onComplete() {
                 GlobalManager.getInstance().getEngine().setScene(GlobalManager.getInstance().getMainScene().getScene());
                 GlobalManager.getInstance().getMainScene().loadBeatmap();
                 initPreferences();
+                availableInternalMemory();
                 if (willReplay) {
                     GlobalManager.getInstance().getMainScene().watchReplay(beatmapToAdd);
                     willReplay = false;
                 }
             }
         });
     }
+    /*
+    Accuracy isn't the best, but it's sufficient enough
+    to determine whether storage is low or not
+     */
+    private void availableInternalMemory() {
+        DecimalFormat df = new DecimalFormat("#.##");
+        df.setRoundingMode(RoundingMode.HALF_EVEN);
+
+        double availableMemory;
+        double minMem = 1073741824D; //1 GiB = 1073741824 bytes
+        File internal = Environment.getExternalStorageDirectory();
+        StatFs stat = new StatFs(internal.getPath());
+        if(Build.VERSION.SDK_INT >= 18) {
+            availableMemory = (double) stat.getAvailableBytes();
+        } else {
+            long blockSize = stat.getBlockSize();
+            long availableBlocks = stat.getAvailableBlocks();
+            availableMemory = (double) (availableBlocks * blockSize);
+        }
+        if(availableMemory < 0.5*minMem) { //I set 512MiB as a minimum
+            Toast.makeText(this, "Storage space is low. Free Storage: " + df.format(availableMemory / minMem)
+                    +  " GB. This may cause replay/score submission bugs.", Toast.LENGTH_SHORT).show();
+        }
```

Comment:
do not use hard coded string

Reply:
oh yeah i forgot, lemme do a quick change